tests/modification-check.py

```diff
@@ -57,7 +57,7 @@ def validate_scripts_content(repo_path, temp_cwd):
             'letsencrypt-auto-source/pieces/fetch.py')), temp_cwd)
 
         # Compare file against current version in the target branch
-        branch = os.environ.get('TRAVIS_BRANCH', 'master')
+        branch = os.environ.get('BUILD_SOURCEBRANCHNAME', 'master')
```

Comment:
I have the same questions about whether or not this variable is properly exposed, but I also don't think its value is quite what we want.

The goal here is to find the target branch of a PR. In most cases this is `master`, but not in the case of release branches or the occasional branch we spin up temporarily like `apache-parser-v2`. We find this branch and allow this test to pass if certbot-auto's contents are either the same as the branch targeted by the PR or if its contents are the latest released version of the script which allows us to submit PRs properly updating its contents.

In Travis, `TRAVIS_BRANCH` refers to the branch targeted by the PR for PR builds. For non-PR builds `TRAVIS_BRANCH` always referred back to the branch itself so it always passed there.

I'm not sure if Azure has something similar, but my understanding is this variable always refers back to the branch/PR itself which isn't what we want for PR builds.

Reply:
OK I got it. Fixes will be quite similar. Azure is exposing some variables specific to pull requests, I think we can use that.

Here also, in an effort to not tie the logic to specific CI provider, I would like to propose the following:
1) Make the target branch a parameter passed to modification-check.py when called, instead of reading an env variable at runtime
2) Feed this value from a generic `TARGET_BRANCH` in the relevant toxenvs (defaulting to master)
3) Set the `TARGET_BRANCH` value appropriately in Azure

Reply:
That SGTM with one tweak:

I'd like to just use `passenv` for the variable in `tox.ini` and continue to set the default value if the variable isn't set in the script itself. I think this avoids coupling that script and `tox` which I think is kind of nice and I don't see any real benefit of us doing so.

One other thing to flag here:

This variable (which you may want to rename due to this) needs to be the tag if the build was triggered by a tag. The documentation for `TRAVIS_BRANCH` is [here](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) which has this behavior.

I was initially wondering if we could simplify things and always use `master` here which I think would work in almost all cases, but one critical case where I think this wouldn't work is on builds triggered by a tag.

Unless you have another idea, I unfortunately think we essentially need to mimic the behavior of `TRAVIS_BRANCH` with the variable you create here.

Reply:
Ok, I did that.